server/src/main/java/com/paypal/selion/proxy/SeLionRemoteProxy.java

```diff
@@ -119,35 +120,57 @@ public SeLionRemoteProxy(RegistrationRequest request, Registry registry) throws
         proxyLogger.info(info.toString());
     }
 
-    /**
-     * Upgrades the node by calling {@link NodeAutoUpgradeServlet} and then {@link #requestNodeShutdown}
-     * 
-     * @param downloadJSON
-     *            the download.json to install on node
-     * @return <code>true</code> on success. <code>false</code> when an error occured.
-     */
-    public boolean upgradeNode(String downloadJSON) {
+    private boolean isSupportedOnNode(Class<? extends HttpServlet> servlet) {
+        LOGGER.entering();
+
         final int TIME_OUT = 30 * 1000;
         RequestConfig config = RequestConfig.custom().setConnectTimeout(TIME_OUT).setSocketTimeout(TIME_OUT).build();
 
         CloseableHttpClient client = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
-        String url = String.format("http://%s:%d/extra/%s", machine, this.getRemoteHost().getPort(),
-                NodeAutoUpgradeServlet.class.getSimpleName());
-        HttpPost post = new HttpPost(url);
-
-        List<NameValuePair> nvps = new ArrayList<NameValuePair>();
-        NameValuePair jsonNVP = new BasicNameValuePair(GridAutoUpgradeDelegateServlet.PARAM_JSON, downloadJSON);
-        nvps.add(jsonNVP);
+        String url = String.format("http://%s:%d/extra/%s", machine, getRemoteHost().getPort(),
+                servlet.getSimpleName());
 
         try {
-            post.setEntity(new UrlEncodedFormEntity(nvps));
-            client.execute(post);
-        } catch (ClientProtocolException e) {
+            HttpGet get = new HttpGet(url);
+            final HttpResponse getResponse = client.execute(get);
+
+            if (getResponse.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+                proxyLogger.warning("Node " + getId() + " does not have or support "
+                        + servlet.getSimpleName());
+                LOGGER.exiting(false);
+                return false;
+            }
+        } catch (IOException e) {
             LOGGER.log(Level.SEVERE, e.getMessage(), e);
+            LOGGER.exiting(false);
             return false;
+        } finally {
+            try {
+                client.close();
+            } catch (IOException e) {
+                LOGGER.log(Level.SEVERE, e.getMessage(), e);
+            }
+        }
+        return true;
+    }
+
+    private HttpResponse sendToNodeServlet(Class<? extends HttpServlet> servlet, List<NameValuePair> nvps) {
+        LOGGER.entering();
+
+        final int TIME_OUT = 30 * 1000;
```

Comment:
We can move this variable to class level, because it is used by more than one method.